src/main/java/org/assertj/core/internal/BinaryDiff.java

```diff
@@ -37,12 +37,18 @@ public BinaryDiffResult diff(File actual, byte[] expected) throws IOException {
 
   @VisibleForTesting
   public BinaryDiffResult diff(Path actual, byte[] expected) throws IOException {
-    try (InputStream expectedStream = new ByteArrayInputStream(expected);
-        InputStream actualStream = Files.newInputStream(actual)) {
-      return diff(actualStream, expectedStream);
+    try (InputStream actualStream = Files.newInputStream(actual)) {
+      return diff(actualStream, expected);
     }
   }
 
+  @VisibleForTesting
+  public BinaryDiffResult diff(InputStream actualStream, byte[] expected) throws IOException {
```

Comment:
Can you add tests for this method in a new test class `BinaryDiff_diff_InputStream_bytes_Test` alongside `BinaryDiff_diff_InputStream_Test` ?